utf8.c

```diff
@@ -1254,39 +1271,51 @@ or both reject and warn.  Rejecting means that the sequence gets translated
 into the Unicode REPLACEMENT CHARACTER instead of what it was meant to
 represent.
 
-Even if a flag is passed that indicates warnings are desired; no warning will be
-raised if C<'utf8'> warnings (or the appropriate subcategory) are disabled at
-the point of the call.
+Unless otherwise stated below, warnings are subject to the C<utf8> warnings
+category being on.
 
 =over 4
 
 =item C<UTF8_CHECK_ONLY>
 
-This also suppresses any warnings.  And it changes what is stored into
+This suppresses any warnings.  And it changes what is stored into
 C<*retlen> with the C<uvchr> family of functions (for the worse).  It is not
 likely to be of use to you.  You can use C<UTF8_ALLOW_ANY> (described below) to
 also turn off warnings, and that flag doesn't adversely affect C<*retlen>.
 
+=item C<UTF8_FORCE_WARN_IF_MALFORMED>
+
+Normally, no warnings are generated if warnings are turned off lexically or
+globally, regardless of any flags to the contrary.  But this flag effectively
+turns on warnings temporarily for the duration of this function's execution.
+
+Do not use it lightly.
+
+This flag is ignored if C<UTF8_CHECK_ONLY> is also set.
+
 =item C<UTF8_DISALLOW_SURROGATE>
 
 =item C<UTF8_WARN_SURROGATE>
 
-These disallow and/or warn about UTF-8 sequences that represent surrogate
-characters.
+These reject and/or warn about UTF-8 sequences that represent surrogate
+characters.  The warning categories C<utf8> and C<super> control if warnings
+are actually raised.
 
 =item C<UTF8_DISALLOW_NONCHAR>
 
 =item C<UTF8_WARN_NONCHAR>
 
-These disallow and/or warn about UTF-8 sequences that represent non-character
-code points.
+These reject and/or warn about UTF-8 sequences that represent non-character
+code points.  The warning categories C<utf8> and C<nonchar> control if warnings
+are actually raised.
 
 =item C<UTF8_DISALLOW_SUPER>
 
 =item C<UTF8_WARN_SUPER>
 
-These disallow and/or warn about UTF-8 sequences that represent code points
-above 0x10FFFF.
+These reject and/or warn about UTF-8 sequences that represent code points
+above 0x10FFFF.  The warning categories C<utf8> and C<super> control if
+warnings are actually raised.
```

Comment:
I don't see the `super` warning category in warnings.pm and this patch doesn't introduce them.

Also below in `UTF8_WARN_PERL_EXTENDED`